src/modules/account/internal/account-param-validation.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  CreateAccountParams,
+  ValidationError,
+  ValidationFailure,
+} from '../types';
+
+export default class AccountParamValidation {
+  public static validate(params: CreateAccountParams): void {
+    const failures: ValidationFailure[] = [];
+    this.isUsernameValid(params.username, failures);
+    this.isPasswordValid(params.password, failures);
+    if (failures.length) {
+      throw new ValidationError(
+        'Account cannot be created, please check the params validity.',
+        failures,
+      );
+    }
+  }
+
+  public static isUsernameValid(
+    username: string,
+    failures: ValidationFailure[],
+  ) {
+    if (username.length < 8) {
+      failures.push({
+        field: 'password',
+        message: 'Username must be at least 8 characters.',
+      });
+    }
+  }
+
+  public static isPasswordValid(
+    password: string,
+    failures: ValidationFailure[],
+  ) {
+    if (password.length < 8) {
```

Comment:
We should not use our own custom validation. Best to use something like https://github.com/dropbox/zxcvbn.

We can use this both on frontend and backend